lib/testutils/resources.go

```diff
@@ -199,8 +199,29 @@ func (t *testResourceWatcher) expectEvents(kind string, anyOrder bool, expectedE
 	// events, so protect against that scenario - we'll check later once we've
 	// constructed useful diagnostics.
 	var actualEvents []watch.Event
+	log.Infof("Received %s events, expected %d", len(t.events), len(expectedEvents))
+	// CASEY: TODO - Can we assert this generally?
+	Expect(len(t.events)).To(Equal(len(expectedEvents)), "Wrong number of events received")
```

Comment:
I'm afraid I don't know if it is correct to assert this generally.  I suggest trying to check out by reviewing where and how `expectEvents` is called.